tests/integration/pkg/systemdnode/systemnode.go

```diff
@@ -22,21 +25,62 @@ func New(clients *clients.Clients, namespace, script string) (*corev1.Pod, error
 		return nil, err
 	}
 
+	imagesPath := fmt.Sprintf("/var/lib/rancher/%s/agent/images", os.Getenv("DIST"))
 	pod := &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:    namespace,
 			GenerateName: "test-node-",
 		},
 		Spec: corev1.PodSpec{
-			Volumes: []corev1.Volume{{
-				Name: "seed",
-				VolumeSource: corev1.VolumeSource{
-					ConfigMap: &corev1.ConfigMapVolumeSource{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: cm.Name,
+			Volumes: []corev1.Volume{
+				{
+					Name: "seed",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: cm.Name,
+							},
 						},
 					},
 				},
+				{
+					Name: "image-assets",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/image-assets",
+						},
+					},
+				},
+				{
+					Name: "agent-images",
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			InitContainers: []corev1.Container{{
```

Comment:
Why is this necessary? Why can't we just mount the `/image-assets` directory into the container (not the init container)?

Reply:
The reason is due to the fact that it appears that RKE2 actually mutates the directory to indicate it wants to pull other files